Add version info for run_fastsurfer and into the docker image

dkuegler opened the pull request:
- Adds extra option to run_fastsurfer.sh: --version with extra option long
  - outputs a version identifier (`{VERSION_TAG}-{git-hash} ({git-branch})`)
  - long option (`--version long`) also adds md5sums of checkpoints and git status to document difference to most recent git commit
- Add VERSION file to docker images
- add version to logfile

dkuegler committed:
Add version info for run_fastsurfer and into the docker image

m-reuter commented:
thanks, but version tags should be read from GitHub (or in the future from the toml file), so that they are only in one place.

dkuegler commented:
sure, and all but the "VERSION_TAG" is read from git, but right now, we do not support this in our git.

dkuegler committed:
- create fastsurfer.toml file
- get version tag from toml file
- create versioned docker images for fastsurfer and tag the newest image as latest

dkuegler committed:
- create fastsurfer.toml file
- get version tag from toml file
- create versioned docker images for fastsurfer and tag the newest image as latest

dkuegler committed:
- fix pyproject.toml file
- update code for version extraction

dkuegler committed:
- fix pyproject.toml file
- update code for version extraction

m-reuter commented on run_fastsurfer.sh:
```diff
@@ -247,20 +247,29 @@ EOF
   if [ "$#" == "1" ] && [ "$1" == "long" ]; then
     pushd "$FASTSURFER_HOME/checkpoints" > /dev/null || return
-    VERSION_INFO=$(printf "%s\ncheckpoints:\n%s" "$VERSION_INFO" "$(md5sum -- *)")
+      VERSION_INFO=$(printf "%s\ncheckpoints:\n%s" "$VERSION_INFO" "$(md5sum -- *)")
```
is this indentation on purpose?

dkuegler replied:
yes, to show the pushd / popd context

dkuegler replied:
but it was pretty inconsistent :(

m-reuter commented on pyproject.toml:
```diff
@@ -0,0 +1,65 @@
+
+[project]
+name = 'fastsurfer'
+version = "2.1.1"
```
Should be 2.2.0-dev.

dkuegler committed:
edit the version number, remove indentation in run_fastsurfer.sh

m-reuter committed:
update version

dkuegler commented:
The disadvantage of 2.2.0-dev is that with git hash this looks a bit more ugly: 2.2.0-dev-62df19 vs. 2.2.0dev-62df19
I think that is why without the extra dash is usually the convention in all open source projects that i like, e.g. pytorch with stuff like 1.14a3 or Firefox with 63.1rc1.

LeHenschel commented on run_fastsurfer.sh:
```diff
@@ -99,7 +100,8 @@ FLAGS:
   --sd  <subjects_dir>    Output directory \$SUBJECTS_DIR (or pass via env var)
-  --t1  <T1_input>        T1 full head input (not bias corrected). Requires an ABSOLUTE Path!
+  --t1  <T1_input>        T1 full head input (not bias corrected). Requires an
+                            ABSOLUTE Path!
```
Why is this on a new line?

LeHenschel replied:
Looks ok to me. Did you test if this works correctly in docker and native install?

dkuegler replied:
Newline because of line length

dkuegler committed:
fix docker_build.sh
write Version to console as well
make log dir before making log
add .gitignore

dkuegler commented:
@LeHenschel finally got the comparison scripts to process everything properly and as I ran both the currently published and this version locally all segmentations and stats files are identical.

It might be that some results differ based on the PC, since @AhmedFaisal95 's sample reference_data has differences to my results.
